### PR TITLE
Refactoring testmodule

### LINKED ISSUE #245: testing modules selectively
"s, test module questions" should run integration tests for that one module only

### PR DESCRIPTION
**Pull request by MatthewBaggins**
from `TestModule.py` docstring

    - You can only test Stampy in the #talk-to-stampy channel
    - You can test Stampy in one of three ways
        1. test all modules; `s, test yourself` or `s, test modules`
        2. test a subset of modules `s, test modules <module-name-1> <module-name-2> <module-name-3>
        3. test one module: `s, test module <module-name>`

Other than that: I refactored the module, made formatting of test messages slightly nicer etc

closes #245

### CODE REVIEW

**Comment by Aprillion** [comment]
FTR: I won't be able to review large PRs for ~a week or so..

File: modules/testModule.py [hunk 1]
```diff
@@ -1,138 +1,285 @@
...
+        3. test one module: `s, test module <module-name>`
+    """
+
+    TEST_PREFIXES = {TEST_MESSAGE_PREFIX, TEST_RESPONSE_PREFIX}
+    TEST_MODULE_PROMPTS = {"test yourself", "test modules", "test module"}
     TEST_PHRASES = {TEST_RESPONSE_PREFIX} | TEST_MODULE_PROMPTS
-    TEST_MODE_RESPONSE_MESSAGE = (
-        "I am running my integration test right now and I cannot handle your request until I am finished"
-    )
+    TEST_MODE_RESPONSE_MESSAGE = "I am running my integration test right now and I cannot handle your request until I am finished"
```
Comment (Lovkush-A) [inline modules/testModule.py:33]: do we have maximum character width? this line might be 'too long'

File: modules/testModule.py [hunk 2]
```diff
@@ -1,138 +1,285 @@
...
+            module_name = next(
+                module_name
+                for module_name in self.utils.modules_dict
+                if module_name.lower() == parsed_module_name.lower()
+            )
+            modules_dict = {module_name: self.utils.modules_dict[module_name]}
+        elif message.clean_content.strip().endswith("test module"):
+            return Response(
+                confidence=10,
+                text="Yeah but what module?",
```
Comment (Lovkush-A) [inline modules/testModule.py:110]:
```suggestion
                text="Yeah but which module?",
```
if you want to be grammatically precise...

**Review by Lovkush-A — CHANGES_REQUESTED** [review]
finally got around to this. sorry for delay.

- minor suggestions in comments below.
- looks like there is bug:
image

- what do you think of combining the two halves of the output together? we currently duplicate all the sent and received messages and (without having looked deeply into the code) i dont know why we do it.

File: modules/testModule.py (hunk 1 above)
Comment (MatthewBaggins) [reply modules/testModule.py:33]: IIRC we use black with standard settings for autoformatting but

**Review by Lovkush-A — CHANGES_REQUESTED** [review]
happy with latest batch of changes. found another bug. `test modules` seems to dislike capital letters.

image

**Review by Lovkush-A — APPROVED** [review]

**Pull request merged**